### PR TITLE
Provide to SCHISM the necessary information on Fortran, CXX and C compiler

### PR DESCRIPTION
**Pull request by ghost**
moved from #4

### CODE REVIEW

**Comment by ghost** [comment]
https://github.com/noaa-ocs-modeling/NEMS/pull/4#issue-673877553 @josephzhang8 @pvelissariou1 @platipodium

File: src/incmake/component_SCHISM.mk [hunk 1]
```diff
@@ -28,19 +40,17 @@ SCHISM_ALL_OPTS= \
...
-	+$(MODULE_LOGIC); echo "SCHISM_SRCDIR = $(SCHISM_SRCDIR)"; exec cmake -S $(SCHISM_SRCDIR) -B $(SCHISM_ROOTDIR)/build -DCMAKE_VERBOSE_MAKEFILE=TRUE
+	+$(MODULE_LOGIC); echo "SCHISM_SRCDIR = $(SCHISM_SRCDIR)"; exec cmake -S $(SCHISM_SRCDIR) -B $(SCHISM_ROOTDIR)/build -DCMAKE_VERBOSE_MAKEFILE=TRUE \
+	 -DCMAKE_Fortran_COMPILER=$(ESMF_F90COMPILER) -DCMAKE_CXX_COMPILER=$(ESMF_CXXCOMPILER) -DCMAKE_C_COMPILER=$(ESMF_CCOMPILER)
    ### Compile the SCHISM components
 	+cd $(SCHISM_BLDDIR); exec $(MAKE) pschism
 #	cd $(SCHISM_BLDDIR); exec $(MAKE) install
-	make -C  $(SCHISM_ROOTDIR)/thirdparty/schism-esmf install-nuopc DESTDIR=$(SCHISM_BINDIR) SCHISM_BUILD_DIR=$(SCHISM_ROOTDIR)/build
-	#+$(MODULE_LOGIC); cd $(SCHISM_SRCDIR)/thirdparty/schism-esmf/src/schism; exec $(MAKE) $(SCHISM_ALL_OPTS) install-nuopc  \
+	make -C  $(SCHISM_ROOTDIR)/../schism-esmf install-nuopc DESTDIR=$(SCHISM_BINDIR) SCHISM_BUILD_DIR=$(SCHISM_ROOTDIR)/build
+	#+$(MODULE_LOGIC); cd $(SCHISM_SRCDIR)/../schism-esmf/src/schism; exec $(MAKE) $(SCHISM_ALL_OPTS) install-nuopc  \
```
Comment (ghost) [inline src/incmake/component_SCHISM.mk]: is this location of `schism-esmf` correct?

**Comment by platipodium** [comment]
No its not. The two dots `..` needs to be replaced by the string `thirdparty`, i.e. 

`$(SCHISM_ROOTDIR)/thirdparty/schism-esmf`

**Comment by ghost** [comment]
@platipodium gets the following error when trying to test this:
```
CMake Error: The source directory "/gpfs/home/lemmen/devel/nems/ADC-WW3-NWM-NEMS/NEMS/ " does not exist.
```
it appears that there is a trailing space in the directory `NEMS/ "`

**Comment by ghost** [comment]
currently testing NEMS+SCHISM build on Hera:
```
/scratch2/COASTAL/coastal/save/shared/repositories/CoastalApp_test
```
```bash
 Zachary.Burnett@hfe03  repositories/CoastalApp_test   feature/schism
 > sh build.sh --plat hera --component SCHISM --compiler intel
```
EDIT: the build fails with an error, I will look into the loaded modules
```
The following variables are defined:
    CLEAN          = 0
    COMPILER       = intel
    NEMS_COMPILER  = intel
    MODULES FILE   = envmodules_intel.hera
    COMPONENTS     = SCHISM
    OS             = linux
    PLATFORM       = hera
    VERBOSE        =

    HDF5HOME       = /scratch1/NCEPDEV/nems/emc.nemspara/soft/netcdf_parallel_release
    NETCDFHOME     = /scratch1/NCEPDEV/nems/emc.nemspara/soft/netcdf_parallel_release
    NETCDF_INCDIR  = /scratch1/NCEPDEV/nems/emc.nemspara/soft/netcdf_parallel_release/include
    NETCDF_LIBDIR  = /scratch1/NCEPDEV/nems/emc.nemspara/soft/netcdf_parallel_release/lib

    ESMFMKFILE     = /scratch1/NCEPDEV/nems/emc.nemspara/soft/esmf/8.0.0-intel18.0.5.274-impi2018.0.4-netcdf4.7.4_parallel/lib/esmf.mk


Currently Loaded Modules:
  1) szip/2.1       3) intel/18.0.5.274   5) hdf5_parallel/1.10.6.release    7) esmf/8.0.0_ParallelNetCDF
  2) cmake/3.16.1   4) impi/2018.0.4      6) netcdf_parallel/4.7.4.release



Are these values correct? [y/n]: y
```
https://github.com/noaa-ocs-modeling/NEMS/files/6726463/build.txt
```
mpiifort -g -traceback -fp-model precise -O2 -fPIC -assume realloc_lhs -m64 -mcmodel=small -threads  -qopenmp -I/scratch1/NCEPDEV/nems/emc.nemspara/soft/esmf/8.0.0-intel18.0.5.274-impi2018.0.4-netcdf4.7.4_parallel/mod -I/scratch1/NCEPDEV/nems/emc.nemspara/soft/esmf/8.0.0-intel18.0.5.274-impi2018.0.4-netcdf4.7.4_parallel/include -I/scratch1/NCEPDEV/nems/emc.nemspara/soft/netcdf_parallel/include -I/scratch2/COASTAL/coastal/save/shared/repositories/CoastalApp_test/SCHISM/build/include -I src/schism  -c src/schism/schism_cmi_nuopc.F90
src/schism/schism_cmi_nuopc.F90(40): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [SCHISM_BMI]
  use schism_bmi
------^
src/schism/schism_cmi_nuopc.F90(41): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [SCHISM_ESMF_UTIL]
  use schism_esmf_util
------^
src/schism/schism_cmi_nuopc.F90(370): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [SCHISM_ESMF_UTIL]
  use schism_esmf_util, only : addSchismMesh
------^
src/schism/schism_cmi_nuopc.F90(481): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [SCHISM_BMI]
  use schism_bmi, only : schismTimeStep
------^
src/schism/schism_cmi_nuopc.F90(514): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [SCHISM_BMI]
  use schism_bmi, only : schismTimeStep
------^
src/schism/schism_cmi_nuopc.F90(253): error #6632: Keyword arguments are invalid without an explicit interface.   [COMMUNICATOR]
  call schism_parallel_init(communicator=mpiCommDuplicate)
----------------------------^
src/schism/schism_cmi_nuopc.F90(467): error #6632: Keyword arguments are invalid without an explicit interface.   [MESH]
      mesh=mesh2d, typeKind=ESMF_TYPEKIND_R8, rc=localrc)
------^
src/schism/schism_cmi_nuopc.F90(467): error #6632: Keyword arguments are invalid without an explicit interface.   [TYPEKIND]
      mesh=mesh2d, typeKind=ESMF_TYPEKIND_R8, rc=localrc)
-------------------^
src/schism/schism_cmi_nuopc.F90(467): error #6632: Keyword arguments are invalid without an explicit interface.   [RC]
      mesh=mesh2d, typeKind=ESMF_TYPEKIND_R8, rc=localrc)
----------------------------------------------^
src/schism/schism_cmi_nuopc.F90(370): error #6580: Name in only-list does not exist or is not accessible.   [ADDSCHISMMESH]
  use schism_esmf_util, only : addSchismMesh
-------------------------------^
src/schism/schism_cmi_nuopc.F90(496): error #6406: Conflicting attributes or multiple declaration of name.   [SCHISMTIMESTEP]
  call schismTimeStep(seconds)
-------^
src/schism/schism_cmi_nuopc.F90(481): error #6580: Name in only-list does not exist or is not accessible.   [SCHISMTIMESTEP]
  use schism_bmi, only : schismTimeStep
-------------------------^
src/schism/schism_cmi_nuopc.F90(529): error #6406: Conflicting attributes or multiple declaration of name.   [SCHISMTIMESTEP]
  call schismTimeStep(seconds)
-------^
src/schism/schism_cmi_nuopc.F90(514): error #6580: Name in only-list does not exist or is not accessible.   [SCHISMTIMESTEP]
  use schism_bmi, only : schismTimeStep
-------------------------^
compilation aborted for src/schism/schism_cmi_nuopc.F90 (code 1)
make[1]: *** [src/schism/schism_cmi_nuopc.o] Error 1
make[1]: Leaving directory `/scratch2/COASTAL/coastal/save/shared/repositories/CoastalApp_test/SCHISM/thirdparty/schism-esmf'
make: *** [/scratch2/COASTAL/coastal/save/shared/repositories/CoastalApp_test/SCHISM_INSTALL/schism.mk] Error 2
```

**Pull request merged**